content/terraform/v1.12.x/docs/language/block/stack/deploy/deployment.mdx

```diff
@@ -0,0 +1,294 @@
+---
+page_title: deployment block reference for Terraform Stack deployments
+description: Use the `deployment` block to define how many times you want to deploy your Stack's infrastructure.
+---
+
+# `deployment` block reference
+
+Use the `deployment` block to define how many times you want to deploy your Stack's infrastructure. You can define `deployment` blocks in a `<NAME>.tfdeploy.hcl` file, which is the deployment configuration file for your Stack.
+
+## Background
+
+<Note>
+
+HCP Terraform supports up to a maximum of 20 deployments per Stack.
+
+</Note>
+
+The `deployment` block defines how to deploy your Stack's infrastructure. Each deployment represents a separate instance of your Stack infrastructure. Each deployment runs in its own isolated agent, wholly separated from other Stack deployments.
+
+Each Stack requires at least one `deployment` block, and you can add additional `deployment` blocks every time you want to deploy your Stack's infrastructure again with different input values. To learn more, refer to [Define deployment configuration](/terraform/language/stacks/deploy/config).
+
+You can define the following configuration blocks in a `tfdeploy.hcl` file:
+- Use the [`deployment_group`](/terraform/language/block/stack/deploy/deployment-auto-group) block to create a group of deployments and apply rules to those deployments.
+- Use the [`deployment_auto_approve`](/terraform/language/block/stack/deploy/deployment-auto-group) block to define rules that automatically approve deployment plans based on specific conditions.
+- Use the [`identity_token`](/terraform/language/block/stack/deploy/identity_token) block to authenticate a deployment with a provider configuration using OIDC.
+- Use the [`store`](/terraform/language/block/stack/deploy/store) block to define key-value secrets in your deployment configuration and use existing variable sets in HCP Terraform.
+- Use the [`locals`](/terraform/language/block/stack/deploy/locals) block to define local values that you can reuse within your deployment configuration.
+- Use the [`publish_output`](/terraform/language/block/stack/deploy/publish_output) block to output values that you want to be available to other Stacks.
+- Use the [`upstream_input`](/terraform/language/block/stack/deploy/upstream_input) block to source inputs from another Stack.
+
+
+## Configuration model
+
+The `deployment` block supports the following arguments:
+
+- [`deployment "<LABEL>"`](#deployment) &nbsp; block
+  - [`inputs`](#inputs) &nbsp; map
+  - [`deployment_group`](#deployment_group) &nbsp; reference to deployment group
+  - [`destroy`](#destroy) &nbsp; boolean
```

Comment:
```suggestion
  - [`destroy`](#destroy) &nbsp; boolean
  - [`import`](#TODO) boolean
```

That `import` argument is used for triggering import runs, and is used by the migrate-from-workspace tool.

Reply:
Ooooh I had no idea about this - thank you SO MUCH for calling this out! I'm adding this as a todo and I'll find out who to talk to on the migrate team.